File: Homeworks/C#2/MethodsHomework/04.AppearanceCount/AppearanceCount.cs
/*
Problem 4. Appearance count

Write a method that counts how many times given number appears in given array.
Write a test program to check if the method is workings correctly.
*/


using System;
using System.Collections.Generic;
using System.Linq;
class AppearanceCount
{
    static int CountElementAppearance(int[] array , int number) 
    {
        int counter = 0;
        for (int i = 0; i < array.Length; i++)
        {
            if (array[i] == number)
            {
                counter++;
            }
            
        }
        return counter;
    }


    static void Main()
    {
        int[] arrayNumbers = { 3, 4, 6, 7, 2, 4, 4, 5 };
        //int[] arrayNumbers = { 1, 2, 2, 7, 2, 4, 3 5 };
        //int[] arrayNumbers = { 6 2, 6, 3, 2, 4, 4, 5 };
        //int[] arrayNumbers = { 10, 4, 9, 7, 8 4, 4, 9 };

        Console.Write("Enter number to search for: ");
        int givenNumber = int.Parse(Console.ReadLine());
        int count = CountElementAppearance(arrayNumbers, givenNumber);
        Console.WriteLine("{0} appears {1} times in the array", givenNumber, count);
    }
}

File: Homeworks/JavaScriptFundamentals/06.Functions/05.AppearanceCount/AppearanceCount.js
/*
 Write a function that counts how many times given number appears in given array.
 Write a test function to check if the function is working correctly.
 */

var jsConsole,
    counter = 0,
    i,
    numbers = [1, 5, 8, 2, 4, 8, 8, 7],
    number = 8;


function countNumberAppears() {
    for (i = 0; i < numbers.length; i += 1) {
        if (numbers[i] === number) {
            counter++;
        }
    }
    return counter;
}

function testFunction(expected){
    return countNumberAppears() === expected;
}
jsConsole.writeLine('The given number ' + number + ' appears '
    + countNumberAppears() + ' time(s).');
jsConsole.writeLine('Is countNumberAppears() function working correctly: '
    + testFunction(counter));

console.log('The given number ' + number + ' appears '
    + countNumberAppears() + ' time(s).');
console.log('Is countNumberAppears() function working correctly: '
    + testFunction(counter));